e2e-tests/testing.md

```diff
@@ -0,0 +1,386 @@
+
+        ```bash
+        virsh console e2e-runner
+        ```
+
+    - The domain will automatically shutdown once the initial setup is complete. You can then start it again with `virsh start e2e-runner`.
+    - Now that the initial setup is done, we need to create a snapshot of this "fresh install" state, so we can revert to it later when installing authd.
+
+        ```bash
+        virsh snapshot-create-as e2e-runner --name fresh-install --reuse-external
+        ```
+
+### 3. Install authd and the brokers
+
+Now that the VM is ready, we need to install authd and the brokers we want to test.
+
+1. Start the VM if it's not already running:
+
+    ```bash
+    virsh start e2e-runner
+    ```
+
+2. Install authd (stable and edge versions)
+   - This can be done either through GUI or by SSH. For simplicity, we will use SSH here.
+
+   - The default username is `ubuntu` and the password is whatever you set in the cloud-init file.
+   - To avoid having to interactively type the password every time, we will use `sshpass` to provide it non-interactively.
+   - The VM uses `socat` to forward the SSH port over VSOCK, so the SSH command will look like this:
+
+    ```bash
+    sshpass -p <vm_password> ssh -o StrictHostKeyChecking=no -o ProxyCommand="socat - VSOCK-CONNECT:1000:22" ubuntu@localhost
+    ```
+
+    - From now on, the command will be referred to as `ssh_vm` for simplicity. You can create an alias for it:
+
+    ```bash
+    alias ssh_vm='sshpass -p <vm_password> ssh -o StrictHostKeyChecking=no -o ProxyCommand="socat - VSOCK-CONNECT:1000:22" ubuntu@localhost'
+    ```
+
+    - Now we can proceed to install authd and the brokers. The following steps need to be repeated for each version of authd we want to test (stable and edge).
+    - In the end of the section, we'll provide a script that automates the whole process.
+
+    Assuming we are installing the edge version of authd:
+    1. Install authd:
+        1. Revert to the "fresh install" snapshot:
+            - And start the domain if the snapshot was taken with the domain powered off.
+
+            ```bash
+            virsh snapshot-revert e2e-runner fresh-install
+            ```
+
+        2. Install authd:
+
+            ```bash
+            ssh_vm "sudo DEBIAN_FRONTEND=noninteractive add-apt-repository -y ppa:ubuntu-enterprise-desktop/authd-edge"
+            ssh_vm "sudo DEBIAN_FRONTEND=noninteractive apt-get install -y authd"
+            ```
+
+        3. Create a snapshot of this state, so we can revert to it later when we proceed to configure the brokers.
+
+            ```bash
+            virsh snapshot-create-as e2e-runner --name authd-edge-installed --reuse-external
+            ```
+
+        4. Repeat the above steps for the stable version of authd, just changing the PPA to `ppa:ubuntu-enterprise-desktop/authd` and naming the snapshot `authd-stable-installed`.
+
+        - Script to automate the above steps for both stable and edge versions of authd:
```

Comment:
Would be useful to mention above that there's a script here. I followed the instructions step by step, so at this point I already have the snapshot for the edge version, so I can't run the script as is

Reply:
The script is mentioned at line 135, but maybe it wasn't evident enough?